mlx/primitives.cpp

```diff
@@ -1338,6 +1338,88 @@ std::pair<array, int> LogicalNot::vmap(
   return {logical_not(inputs[0], stream()), axes[0]};
 }
 
+std::vector<array> LogicalAnd::vjp(
+    const std::vector<array>& primals,
+    const array& cotan,
+    const std::vector<int>& argnums) {
+  assert(primals.size() == 2);
+
+  if (argnums.size() == 1) {
+    // If derivative is with respect to only one argument
+    return {argnums[0] == 0 ? cotan : zeros_like(cotan, stream())};
+  } else {
+    // If derivative is with respect to both arguments
+    return {zeros_like(cotan, stream()), zeros_like(cotan, stream())};
+  }
+}
+
+array LogicalAnd::jvp(
+    const std::vector<array>& primals,
+    const std::vector<array>& tangents,
+    const std::vector<int>& argnums) {
+  assert(primals.size() == 2);
+  assert(argnums.size() <= 2);
+
+  if (argnums.size() == 1) {
+    // If derivative is with respect to only one argument
+    return argnums[0] == 0 ? tangents[0] : zeros_like(tangents[0], stream());
+  } else {
```

Comment:
I'm not sure what's the intention here. I don't think it's correct to return the `tangents`. The `jvp` always returns one value. Since it's zeros in this case, you can ignore the `argnums`.